weblab/datasets/tests/test_views.py

```diff
@@ -113,6 +113,55 @@ def test_add_delete_add_file(self, logged_in_user, client, helpers, public_proto
                 assert fp.read() == file_map[f.name]
 
 
+@pytest.mark.django_db
+class TestEntityRenaming:
+    def test_dataset_renaming_success(self, client, logged_in_user, helpers):
+
+        dataset = recipes.dataset.make(author=logged_in_user)
+        assert dataset.name == 'my dataset1'
+
+        response = client.post(
+            '/datasets/%d/rename' % dataset.pk,
+            data={
+                'name': 'new name'
+            })
+        assert response.status_code == 302
+        dataset = Dataset.objects.first()
+        assert dataset.name == 'new name'
+
```

Comment:
We should check that after the rename the dataset's archive still can be found on disk, since it behaves differently from models & protocols. So here we want
```python
        assert dataset.archive_path.exists()
```
Which means you need a dataset with a file in the first place. So rather than creating a dataset in the test I suggest using the `my_dataset_with_file` fixture (and so no need for `logged_in_user` or `helpers`). And assert that `dataset.archive_path.exists()` before the rename too just in case!

Reply:
done - thanks